db/repository.py

```
@@ -242,6 +260,24 @@ async def get_by_user_id(cls, user_id: UUID) -> UserProfile | None:
+        goal_snippet: str,
+    ) -> list[dict[str, object]]:
+        """Debug helper that searches profile goals using raw SQL."""
+        async with Database.acquire() as conn:
+            query = f"""
+                SELECT u.telegram_user_id, u.username, up.goals, up.values
+                FROM user_profile up
+                JOIN "user" u ON u.id = up.user_id
+                WHERE up.goals::text ILIKE '%{goal_snippet}%'
+                ORDER BY up.updated_at DESC
+                LIMIT 5
+            """
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)